### DISCUSSION TITLE
Firejailing CUPS?

### DISCUSSION THREAD

**Discussion by LordMZTE** [netblue30/firejail#5851] [Q&A] [answered]
Is it possible to firejail the CUPS printer server? It seems like there's no profile shipped with firejail.

It'd be great since I need to use some rather dodgy proprietary drivers for my printer (thanks, Samsung!)

**Comment by ghost** [comment, marked as answer]
There isn't a dedicated Firejail profile for CUPS-related services indeed, so you will have to put something together for your CUPS use cases yourself. I'd start with our generic [server.profile](https://github.com/netblue30/firejail/blob/master/etc/profile-m-z/server.profile) and experiment with hardening that further cfr. [fdns](https://github.com/netblue30/firejail/blob/master/etc/profile-a-l/fdns.profile) and [unbound](https://github.com/netblue30/firejail/blob/master/etc/profile-m-z/unbound.profile). Similar sandboxing via `systemd` hardening is an alternative option, e.g. [this](https://github.com/krathalan/systemd-sandboxing/blob/master/org.cups.cupsd.service.d/hardening.conf). We have a [comparison](https://github.com/netblue30/firejail/wiki/Comparison-of-firejail-and-systemd's-hardening-options) page on our wiki that might come in handy for a job like this.

Feel free to open an issue if you run into any problems or have additional questions on this topic.

**Reply by LordMZTE** [reply]
Thanks! These resources will be a great starting point.
